packages/components/src/ui/context/polymorphic-component.ts

```diff
@@ -32,7 +32,7 @@ export type PolymorphicComponent< T extends As, O > = {
 	): JSX.Element | null;
 	( props: ViewOwnProps< O, T > ): JSX.Element | null;
 	displayName?: string;
-	selector: string;
+	selector: `.${ string }`;
```

Comment:
What is the reason for forcing the selector to always be a class selector?

Reply:
🤔 I hadn't really considered that it could be an attribute base selector, but currently it is a class selector 🤷 Do you think this is too inflexible? 99.9999% of components will not declare this on their own, FWIW, View is a huge edge case in this regard.

Reply:
I don't have enough context to know exactly how `selector` is meant to be used, or how many components are going to specify a value.

I was just highlighting that, when seeing a prop called `selector`, my expectation is for it to basically be `Parameters< typeof document.querySelector >`.

If we need to force it to be only a class-based selector, we should at least document it in the code / docs?

Reply:
It's used for component interpolation, so for the following:

```js

const Text = createComponent( { useHook: useText, name: 'Text, as: 'span' } );

const Wrapper = styled.div`
  ${ Text.selector } {
    color: blue;
  }
`;
```

Because the Emotion babel plugin only supports interpolating styled components, we needed this workaround to be able to fake interpolation of components that are part of the component system but outside of styled components.

The reason it's forced to be a class based selector is because it's dependent on the automagically generated classes that are added to components when they're run through `useContextSystem`. That same class is applied as the `selector` value by `contextConnect`. It follows the format of `components-component-name` which is the pre-established pattern in the `@wordpress/components` package.

Making it the same type as `Parameters<typeof document.querySelector>` I think would be misleading, considering it's an auto-generated property. I think encoding it as specifically a class-based selector makes sense given the use case.

Where do you think this information would be best saved? I'm happy to add some documentation around it but it's sort of an internal thing rather than something we wanted to expose publicly.

Reply:
Thank you for the explanation! With this context in mind, I see how it makes sense for this selector to be restricted to a class.

> Where do you think this information would be best saved? I'm happy to add some documentation around it but it's sort of an internal thing rather than something we wanted to expose publicly.
 
Maybe the best place would be with an inline JSDoc snippet? (feel free to re-write the snippet to a more clear version):

```suggestion
	/**
	 * A CSS selector used for component interpolation when the component is outside of Styled Components.
	 * The selector must be class-based, since it depends on other classnames generated by Styled Components.
	 */
	selector: `.${ string }`;
```

Reply:
https://github.com/WordPress/gutenberg/pull/31555/commits/d7f2686c6805a4f6d6ce83b38dc6a75d5079e230 looks great!